File: seed/ruby-sdk-v2/server-sent-event-examples/lib/seed/completions/types/group_deleted_event.rb
# frozen_string_literal: true

module Seed
  module Completions
    module Types
      class GroupDeletedEvent < Internal::Types::Model
        field :offset, -> { String }, optional: false, nullable: false

        field :group_id, -> { String }, optional: false, nullable: false
      end
    end
  end
end

File: seed/php-sdk/server-sent-event-examples/src/Completions/Types/GroupDeletedEvent.php
<?php

namespace Seed\Completions\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class GroupDeletedEvent extends JsonSerializableType
{
    /**
     * @var string $offset
     */
    #[JsonProperty('offset')]
    public string $offset;

    /**
     * @var string $groupId
     */
    #[JsonProperty('group_id')]
    public string $groupId;

    /**
     * @param array{
     *   offset: string,
     *   groupId: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->offset = $values['offset'];
        $this->groupId = $values['groupId'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
